### PR TITLE
Add support for whitelisting non-expiring tokens

### PR DESCRIPTION
**Pull request by charlie47**
…along with whitelisting newsrooms

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -50,7 +57,15 @@ const registerJwtStrategy = function(server, options) {
             server.auth.strategy('jwt', 'jwt', {
                 key: base64toPem(options.publicKey),
                 validateFunc: (decoded, request, callback) => {
-                    callback(null, true);
+                    if (!decoded.exp) {
+                        const nonExpiringIds = options.nonExpiringIds;
+                        const isValid = nonExpiringIds && decoded.jti && nonExpiringIds.indexOf(decoded.jti) > -1;
+                        callback(null, isValid);
+                    } else if (options.newsrooms && options.newsrooms.indexOf(decoded.newsroom) === -1) {
```
Comment (kubek2k) [inline index.js]: there are some voices that this was used to check for newsrooms in relations-api, but to be honest this check doesn't hold anymore. I vote for removing this.

Comment (kubek2k) [reply index.js]: please let me know if you have any idea why this should remain
@pdziok @luki-

File: index.js [hunk 2]
```diff
@@ -1,12 +1,19 @@
...
+ */
 const jwtAuth = require('hapi-auth-jwt2');
 const bearerAuth = require('hapi-auth-bearer-token');
 const Joi = require('joi');
 
 /**
  * @typedef {Object} JwtOptions
  * @property {string} publicKey
+ * @property {Array.<string>} nonExpiringIds Non expiring Id's list
+ * @property {Array.<string>} newsrooms Newsroom white list
```
Comment (pdziok) [inline index.js]: maybe `whitelist`?

File: index.js (hunk 1 above)
Comment (pdziok) [inline index.js]: yo, how about https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/includes?

File: README.md [hunk 3]
```diff
@@ -17,7 +19,9 @@ server.register({
     register: commonAuth,
     options: {
         jwt: {
-            publicKey: '...'
+            publicKey: 'public-key',
+            nonExpiringIds: ['allowed-id-1', 'allowed-id-2'], // optional
+            newsrooms: ['allowed-newsroom-1', 'allowed-newsroom-2'] // optional
```
Comment (luki-) [inline README.md]: This should be removed too, shouldn't it?

Comment (charlie47) [reply README.md]: That's right!

File: README.md [hunk 4]
```diff
@@ -17,7 +19,8 @@ server.register({
     register: commonAuth,
     options: {
         jwt: {
-            publicKey: '...'
+            publicKey: 'public-key',
+            nonExpiringIds: ['allowed-id-1', 'allowed-id-2'], // optional
```
Comment (luki-) [inline README.md]: There's trailing comma left.

File: schemas.js [hunk 5]
```diff
@@ -4,7 +4,9 @@ const Joi = require('joi');
 
 module.exports = {
     jwt: Joi.object({
-        publicKey: Joi.string().required()
+        publicKey: Joi.string().required(),
+        nonExpiringIds: Joi.array().items(Joi.string()).unique(),
+        newsrooms: Joi.array().items(Joi.string()).unique()
```
Comment (luki-) [inline schemas.js]: Here's another leftover

Comment (charlie47) [reply schemas.js]: Good point!

**Review by luki- — APPROVED** [review]

**Pull request merged**